### PR TITLE
Update ESLint config for version 1.5.1 compatibility

### PR DESCRIPTION
**Pull request by anselmbradford**
Update ESLint config for  version 1.5.1 compatibility.
Fixes ESLint errors that were breaking on Travis.

## Additions

- Added additional environments and rules for version 1.5.1 ESLint compatibility.

## Changes

- Fixes linting errors:
  - Formats tests to fix indenting errors.
  - Moves requires to top of files in tests.
- Moves gulp linting implementation to a helper function.

## Testing

- `gulp build && gulp lint` should have no errors

## Review

- @jimmynotjim 
- @sebworks 
- @KimberlyMunoz 

## Todos

- ~~I have a PR open with gulp-eslint for updating that to [1.5.1](https://github.com/adametry/gulp-eslint/pull/96), but this doesn't appear to add any breaking changes so should be able to go in before that is merged.~~ Not necessary, the caret versioning actually already brings in `1.5.1` without an updated needed in gulp-eslint.

### CODE REVIEW

**Comment by jimmynotjim** [comment]
Well somehow we've jumped to 78 warnings, but at least the errors are gone! :+1:

File: test/unit_tests/modules/util/expanded-state-spec.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-  var $, es, sandbox, divExpanded, divClosed, openMenu;
+  var sandbox, divExpanded, divClosed, openMenu;
 
   jsdom();
 
+  $ = require( 'jquery' );
+
   before( function() {
-    $ = require( 'jquery' );
-    es = require( '../../../../src/static/js/modules/util/expanded-state.js' );
```
Comment (KimberlyMunoz) [inline test/unit_tests/modules/util/expanded-state-spec.js]: I thought jsdom required this set up. Lemme quickly test this.

Comment (KimberlyMunoz) [inline test/unit_tests/modules/util/expanded-state-spec.js]: Getting lots of `Error: jQuery requires a window with a document` with these changes.

Comment (anselmbradford) [inline test/unit_tests/modules/util/expanded-state-spec.js]: Yeah, sorry discovered this in https://github.com/cfpb/cfgov-refresh/pull/1007#discussion_r40253020 Pull down the latest.

**Comment by KimberlyMunoz** [comment]
That cf_notifier unit test error was there already, right?

**Comment by anselmbradford** [comment]
> That cf_notifier unit test error was there already, right?

Are you sure you have the latest? I was getting that but can't repeat it now after the latest commit.

**Comment by KimberlyMunoz** [comment]
> Are you sure you have the latest? I was getting that but can't repeat it now after the latest commit

Yeah. It seems to be okay now. Weird. :+1:

**Pull request merged**